assembl/views/api2/auth.py

```diff
@@ -287,6 +301,186 @@ def verify_password(request):
 
 
 @view_config(
+    context=CollectionContext, ctx_collection_class=AgentProfile,
+    request_method='POST', permission=NO_PERMISSION_REQUIRED,
+    name="password_reset", header=FORM_HEADER)
+@view_config(
+    context=ClassContext, ctx_class=AgentProfile, header=FORM_HEADER,
+    request_method='POST', permission=NO_PERMISSION_REQUIRED,
+    name="password_reset")
+def reset_password(request):
```

Comment:
Also see my response to the next comment. Basically, we tried to do a gradual shift, where both mechanisms could coexist, and then shifted to doing a more brutal shift, but did not yet finish killing old code. So yes, you are correct, there is duplication. I do not believe this PR to be complete in that way. As it is, it does add technical debt.